Scripts/ServiceFabricRPHelpers/ServiceFabricRPHelpers.psm1

```diff
@@ -407,9 +407,11 @@ if($CreateSelfSignedCertificate)
 
     Write-Host "Creating new self signed certificate at $NewPfxFilePath"
     
-    ## Changes to PSPKI version 3.5.2 New-SelfSignedCertificate replaced by New-SelfSignedCertificateEx
+    ## Changes to PSPKI version 3.2.5 New-SelfSignedCertificate replaced by New-SelfSignedCertificateEx
     $PspkiVersion = (Get-Module PSPKI).Version
-    if($PSPKIVersion.Major -ieq 3 -And $PspkiVersion.Minor -ieq 2 -And $PspkiVersion.Build -ieq 5) {
```

Comment:
Notice type here, PSPKIVersion is used instead of PspkiVersion

Reply:
Also - when PSPKI  latest version is used (3.2.6) - this condition returns false, thus New-SelfsignedCertificate is used instead of New-SelfsignedCertificateEx. Which is a big problem since Service Fabric services fail (with 'failed to read private key from certificate') to start when using a cert that was created via New-SelfsignedCertificate.

Reply:
Thankyou @aviade